ipalib/rpc.py

```diff
@@ -586,12 +587,31 @@ def __init__(self, *args, **kwargs):
         self.ccache = kwargs.pop("ccache", None)
 
     def _handle_exception(self, e, service=None):
+        def try_acquire_cred():
```

Comment:
IMHO the code would be more readable if you move the function from local scope to a private method `_try_acquire_cred`.

Reply:
Thanks, addressed.